pkg/controllers/status/work_status_controller.go

```diff
@@ -271,11 +276,24 @@ func (c *WorkStatusController) recreateResourceIfNeeded(work *workv1alpha1.Work,
 		}
 
 		desiredGVK := schema.FromAPIVersionAndKind(manifest.GetAPIVersion(), manifest.GetKind())
-		if reflect.DeepEqual(desiredGVK, workloadKey.GroupVersionKind()) &&
+		if !(reflect.DeepEqual(desiredGVK, workloadKey.GroupVersionKind()) &&
```

Comment:
I don't understand what this expression means? Shouldn't it be like the following?
```
if !reflect.DeepEqual(desiredGVK,workloadKey.GroupVersionKind()) || manifest.GetNamespace() != workloadKey.Namespace || manifest.GetName() != workloadKey.Name)
```